vowpalwabbit/slim/src/CMakeLists.txt

```diff
@@ -0,0 +1,7 @@
+find_package(Boost)
```

Comment:
Also needs to be  converted to new style cmake

Reply:
Can you elaborate?  What changes are needed for new style CMake?

---
In reply to: [270508510](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1800#discussion_r270508510) [](ancestors = 270508510)

Reply:
See how we consume boost for the main vw target

Reply:
Looked at the main vw target.  The difference here is that a preprocessor macro is used to enable code that uses boost if boost is found.  I don't see a target version of add_definitions.  Is there a target_add_definitions?  Open to suggestions.

---
In reply to: [304041869](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1800#discussion_r304041869) [](ancestors = 304041869)

Reply:
Looked at the main vw target.  The difference here is that a preprocessor macro is used to enable code that uses boost if boost is found.  I don't see a target version of add_definitions.  Is there a target_add_definitions?  Open to suggestions.

---
In reply to: [303967273](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1800#discussion_r303967273) [](ancestors = 303967273,270508510)

Reply:
Thanks for fixing it Jack

---
In reply to: [304440622](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1800#discussion_r304440622) [](ancestors = 304440622,304041869)

Reply:
Thanks for fixing it Jack

---
In reply to: [304441060](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1800#discussion_r304441060) [](ancestors = 304441060,303967273,270508510)